lib/jsdom/living/nodes/HTMLElement-impl.js

```diff
@@ -58,6 +58,14 @@ class HTMLElementImpl extends ElementImpl {
     }
   }
 
+  dispatchEvent(eventImpl) {
+    if (isDisabled(this)) {
+      return;
+    }
+
+    return super.dispatchEvent(eventImpl);
```

Comment:
This is not correct. https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#enabling-and-disabling-form-controls:-the-disabled-attribute:concept-fe-disabled says:

> A form control that is disabled must prevent any `click` events that are queued on the user interaction task source from being dispatched on the element.

So any special processing must only apply to `click` events.

Additionally, some places call `_dispatch` directly: https://github.com/jsdom/jsdom/blob/e3744f5ceffbf06a83b8e0c11b1641f4e229ea18/lib/jsdom/living/helpers/events.js#L17 It might be better to override `_dispatch` instead of the more public `dispatchEvent`.

Reply:
That can't be the right part of the spec, because the user interaction task source is never involved in jsdom, I don't think.

The click() disabled handling is in the spec at https://html.spec.whatwg.org/multipage/interaction.html#activation:concept-fe-disabled.

For dispatchEvent, I cannot find any support in the spec for this. Furthmore, running the test in #2665 in Firefox shows that it matches jsdom.

So I don't think we should change this in jsdom, but there is a spec interop issue, apparently.

Reply:
This is https://github.com/whatwg/html/issues/2368